neural_network/2_hidden_layers_neural_network.py

```diff
@@ -0,0 +1,203 @@
+import numpy
+
+
+class NeuralNetwork:
+    def __init__(self, input_array: numpy.array, output_array: numpy.array) -> None:
+        """
+        input_array : input values for training the neural network.
+        output_array : expected output values of the given inputs.
+        """
+        self.input = input_array
+        # Initial weights are assigned randomly where first argument is the number
+        # of nodes in previous layer and second argument is the
+        # number of nodes in the next layer.
+
+        # random initial weights for the input layer
+        # self.input.shape[1] is used to represent number of nodes in input layer
+        # first hidden layer consists of 4 nodes
+        self.weights1 = numpy.random.rand(self.input.shape[1], 4)
```

Comment:
Instead of weights1, weights2,... we should provide descriptive names such that when we're referencing them in other parts of the class they are easy to pick up. Maybe it should be named: `input_layer_weights, first_layer_weights, etc`. You can change it to whatever suits you best.